### PR TITLE
Reduce initial user query load

### PR DESCRIPTION
**Pull request by BerkeleyTrue**
- Refactor passport to own file
- Cache hot news
- deprecate news creation api

### CODE REVIEW

File: server/component-passport.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  // enabled
+
+  passport.serializeUser((user, done) => {
+    done(null, user.id);
+  });
+
+  passport.deserializeUser((id, done) => {
+
+    this.userModel.findById(id, { fields }, (err, user) => {
+      if (err || !user) {
```
Comment (BerkeleyTrue) [inline server/component-passport.js]: add field filter to exclude potentially large user properties

File: server/component-passport.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        return done(err, user);
+      }
+      this.app.dataSources.db.connector
+        .collection('user')
+        .aggregate([
+          { $match: { _id: user.id } },
+          { $project: { points: { $size: '$progressTimestamps' } } }
+        ], function(err, { points = 1 } = {}) {
+          if (err) { return done(err); }
```
Comment (BerkeleyTrue) [inline server/component-passport.js]: Use MongoDB pipeline aggregation to return user brownie points.

This is done in such a way that we can later change progressTimestamps to scaled brownie points

**Comment by bugron** [comment]
@BerkeleyTrue what's the status of this PR?

**Comment by BerkeleyTrue** [comment]
Now live on http://beta.freecodecamp.com/

**Pull request merged**